src/main/java/uk/gov/hmcts/reform/sscs/services/CaseLoaderService.java

```diff
@@ -223,4 +229,24 @@ private void processCase(SscsCaseData caseData) {
         }
     }
 
+    private List<SscsCaseDetails> getCasesByCaseRefOrCaseIdWithInvalidReferenceErrorHandlingIfEnabled(
+        SscsCaseData caseData,
+        IdamTokens idamTokens
+    ) {
+        if (invalidCaseRefErrorHandlingEnabled) {
+            try {
+                return searchCcdCaseService.findListOfCasesByCaseRefOrCaseId(caseData, idamTokens);
+            } catch (FeignException.FeignClientException feignException) {
+                if (HttpStatus.BAD_REQUEST.value() == feignException.status()) {
+                    log.error(logPrefixWithFile + "FeignException with message {} for case with SC {} and ccdID {} ",
+                        feignException.getMessage(), caseData.getCaseReference(), caseData.getCcdCaseId());
+                    return Collections.emptyList();
+                }
+                throw feignException;
+            }
+        }
+        return searchCcdCaseService.findListOfCasesByCaseRefOrCaseId(caseData, idamTokens);
```

Comment:
is this simpler, or is it less clear
`  try {
           return searchCcdCaseService.findListOfCasesByCaseRefOrCaseId(caseData, idamTokens);
   } catch (FeignException.FeignClientException feignException) {
          if (invalidCaseRefErrorHandlingEnabled && HttpStatus.BAD_REQUEST.value() == feignException.status()) {
             log.error(logPrefixWithFile + "FeignException with message {} for case with SC {} and ccdID {} ",
                        feignException.getMessage(), caseData.getCaseReference(), caseData.getCcdCaseId());
             return Collections.emptyList();
          }
          throw feignException;
  }`